### ISSUE TITLE
[Postgres] unparseable error on `character varying` type

### ISSUE THREAD

**Issue by ackintosh** [sqlfluff/sqlfluff#733]
-- If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant. Finally, please add any labels that are related to the issue before submitting so that it can be addressed easily! --

## Expected Behaviour

```sql
CREATE TABLE main.test_table (
    "col1" character varying(40) NOT NULL
);
```

Linting the DDL statement would be success.

## Observed Behaviour

The linting would be failed.

```bash
$ sqlfluff lint --dialect postgres postgres/table/test.sql
== [postgres/table/test.sql] FAIL
L:   1 | P:   1 |  PRS | Found unparsable section: 'CREATE TABLE main.test_table
                       | (\n "col1...'
```

If we update the DDL to use `varchar` instead linting will success.

```diff
CREATE TABLE main.test_table (
-    "col1" character varying(40) NOT NULL
+    "col1" varchar(40) NOT NULL
);
```

## Version

```bash
$ sqlfluff --version
sqlfluff, version 0.3.6

$ python -V
Python 3.9.1
```

## Configuration

No configurations, using default.
